packages/runtime/src/lib/constants.ts

```diff
@@ -28,14 +31,18 @@ const zeroAddress = new Uint8Array(32);
 const zeroUint64 = 0n;
 const zeroByte = new Uint8Array(0);
 
+// keys with value as null does not represent a txn/global field, these are handled explicitly
+// in txn.ts using switch
+type keyOfEncTx = keyof TxnEncodedObj | keyof AssetDefEnc | keyof StateSchemaEnc;
+
 // https://developer.algorand.org/docs/reference/teal/opcodes/#txn
 // transaction fields supported by teal v1
-export const TxnFields: {[key: number]: {[key: string]: any}} = {
+export const TxnFields: {[key: number]: {[key: string]: keyOfEncTx | null }} = {
   1: {
     Sender: 'snd',
     Fee: 'fee',
     FirstValid: 'fv',
-    FirstValidTime: '',
+    FirstValidTime: null,
```

Comment:
should we use `null` or undefined? I think undefined is better.

Reply:
Actually we can't use undefined here, because in the `op.execute(..)` we want to assert that we tx field passed by user is a valid one (eg. *FirstValidTime* is correct, but *firstvalidtime* isn't).
We check this by `txn[tealv][field] !== undefined`, so if we can't keep `undefined` as a value in our original maps